projects/components/src/skeleton/skeleton.component.ts

```diff
@@ -0,0 +1,85 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+
+@Component({
+  selector: 'ht-skeleton',
+  template: `
+    <ng-container *ngFor="let k of this.numberOfIterations">
+      <ng-container [ngSwitch]="this.shapeStyle">
+        <div *ngSwitchCase="'${SkeletonType.RectangleText}'" [ngClass]="this.containerClass"></div>
+        <div *ngSwitchCase="'${SkeletonType.Circle}'" [ngClass]="this.containerClass"></div>
+        <div *ngSwitchCase="'${SkeletonType.Square}'" [ngClass]="this.containerClass"></div>
+        <div *ngSwitchCase="'${SkeletonType.TableRow}'" [ngClass]="this.containerClass"></div>
+        <div *ngSwitchCase="'${SkeletonType.Donut}'" [ngClass]="this.containerClass">
+          <div class="donut-inner"></div>
+        </div>
+        <div *ngSwitchCase="'${SkeletonType.ListItem}'" [ngClass]="this.containerClass">
+          <div class="item-circle"></div>
+          <div class="item-column">
+            <div class="item-line"></div>
+            <div class="item-line"></div>
+          </div>
+        </div>
+        <div *ngSwitchDefault class="skeleton skeleton-rectangle"></div>
+      </ng-container>
+    </ng-container>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./skeleton.component.scss']
+})
+export class SkeletonComponent implements OnChanges {
+  @Input()
+  public shapeStyle: SkeletonType = SkeletonType.Rectangle;
+
+  @Input()
+  public repeat: number | undefined = 1;
```

Comment:
If we can repeat this with css (may be try background-repeat: repeat-y), then
1. Remove RepeatCount
2. Introduce a local variable "repeatInfinite" which will only be set to true for Table and List. Otherwise, false.

Reply:
If not, then I will recommend removing `repeat` as an input and we can set it to reasonable default like 4 or 5 and only set it when it is  a table or list.